gluten-core/src/main/scala/io/glutenproject/execution/HashJoinExecTransformer.scala

```diff
@@ -179,14 +181,84 @@ trait HashJoinLikeExecTransformer
     case BuildRight =>
       joinType match {
         case _: InnerLike | LeftOuter | LeftSemi | LeftAnti | _: ExistenceJoin =>
-          left.outputPartitioning
+          expandPartitioning(left.outputPartitioning)
         case RightOuter => right.outputPartitioning
         case x =>
           throw new IllegalArgumentException(
             s"HashJoin should not take $x as the JoinType with building right side")
       }
   }
 
+  // https://issues.apache.org/jira/browse/SPARK-31869
+  // ToDo: https://issues.apache.org/jira/browse/SPARK-45882
```

Comment:
I have added a ToDO here, for this https://issues.apache.org/jira/browse/SPARK-45882 we would need to add in shim. Shall that be taken in separate PR?

Reply:
@ulysses-you Shall we move this to shim module as part of this in separate PR?

Reply:
I'd like to solve this todo in this pr, as we need to move the whole code to shim module.

Reply:
Thanks @ulysses-you I will update the PR